src/java/main/org/apache/zookeeper/server/NettyServerCnxnFactory.java

```diff
@@ -109,6 +110,20 @@ public void channelConnected(ChannelHandlerContext ctx,
                     zkServer, NettyServerCnxnFactory.this);
             ctx.setAttachment(cnxn);
 
+            InetAddress addr = ((InetSocketAddress)cnxn.channel.getRemoteAddress()).getAddress();
+            Set<NettyServerCnxn> s = ipMap.get(addr);
```

Comment:
Not sure if this is an issue or not, but if multiple connections (from a new address) hit this line before reaching the next one, the maxclientcnxns will never be checked as factory.ipMap.get(address) will return null for all of the connections?